pkg/ee/metering/handler.go

```diff
@@ -175,19 +172,19 @@ func CreateOrUpdateCredentials(ctx context.Context, request interface{}, seedsGe
 		Endpoint:  []byte(req.Endpoint),
 	}
 
-	for _, client := range seeds {
-		if err := createOrUpdateMeteringToolSecret(ctx, client, data); err != nil {
+	for seed, client := range seeds {
+		if err := createOrUpdateMeteringToolSecret(ctx, client, seed.Namespace, data); err != nil {
```

Comment:
Why you are not using the `k8c.io/kubermatic/v2/pkg/resources.KubermaticNamespace` ?